src/utils/UWebSocketTransport.ts

```diff
@@ -35,80 +43,159 @@ export async function createUWebSocketTransport<T extends { isConnected: boolean
   socket: IUWebSocket<T>,
   uServerEmitter: Emitter<IUWebSocketEventMap>,
   config: IConfigComponent,
-  logs: ILoggerComponent
+  logs: ILoggerComponent,
+  metrics: IMetricsComponent<MetricsDeclaration>
 ): Promise<Transport> {
   const logger = logs.getLogger('ws-transport')
   const transportId = randomUUID()
 
   const maxQueueSize = (await config.getNumber('WS_TRANSPORT_MAX_QUEUE_SIZE')) || 1000
-  const queueDrainTimeoutInMs = (await config.getNumber('WS_TRANSPORT_QUEUE_DRAIN_TIMEOUT_IN_MS')) || 5000
-
-  const messageQueue: Array<{ message: Uint8Array; future: IFuture<void> }> = []
+  const retryDelayMs = (await config.getNumber('WS_TRANSPORT_RETRY_DELAY_MS')) || 1000
+  const maxRetryAttempts = (await config.getNumber('WS_TRANSPORT_MAX_RETRY_ATTEMPTS')) || 5
+  const maxBackoffDelayMs = (await config.getNumber('WS_TRANSPORT_MAX_BACKOFF_DELAY_MS')) || 30000 // 30 seconds maximum delay
 
   let isTransportActive = true
   let isInitialized = false
+
+  // Simple queue for messages that couldn't be sent immediately
+  const messageQueue: Array<{
```

Comment:
Should we keep the message queue even if we have the backpressure buffer? That is, should we keep trying to send messages after the backpressure buffer got as big as we consider it to be it's maximum size?
We can leave it for the time being, but it's something to consider.